nautobot/docs/models/dcim/device.md

```diff
@@ -15,3 +15,12 @@ Device names must be unique within a site, unless the device has been assigned t
 When a device has one or more interfaces with IP addresses assigned, a primary IP for the device can be designated, for both IPv4 and IPv6.
 
 For Devices forming a group (Failover, Load-Sharing, Redundacy or similar) refer to [Device Redundancy Groups](deviceredundancygroup.md) model documentation.
+
+## Developer API
+
+The `Device` Django model class supports a method called `create_components()`. This method is normally called during `Device.save()`, which is called whenever you save create a Device via the GUI or the REST API, but if you are working directly in the ORM and encounter one of the two following scenarios, `Device.save()` is not called:
+
+- Usage of `Device.objects.bulk_create()` to perform a bulk creation of Device objects
+- Usage of `Device.save()` during handling of the `nautobot_database_ready` signal (which uses [historical models](https://docs.djangoproject.com/en/3.2/topics/migrations/#historical-models))
+
+In these cases you will have to manually run `Device.create_components()` in order to instantiate the [device type's](devicetype.md) component templates (interfaces, power ports, etc.).
```

Comment:
My only objection to this is the documentation reads that this is a class function, not an instance function. Could you instead do something like `device_instance.create_components()`?

Reply:
Done!